cw/tvla.py

```diff
@@ -504,9 +535,12 @@ def run_tvla(ctx: typer.Context):
                 log.info("Converting Traces")
                 if project.waves[0].dtype == 'uint16':
                     traces = np.empty((num_traces, num_samples), dtype=np.uint16)
+                    log.info(f"Will use samples from {sample_start} to {sample_start+num_samples}")
                     for i_trace in range(num_traces):
-                        traces[i_trace] = project.waves[i_trace + trace_start]
-                else:
+                        traces[i_trace] = project.waves[i_trace +
+                                                        trace_start][sample_start:sample_start +
+                                                                     num_samples]
+                else:  # FIXME: The wave slicing is only tested for otbn
```

Comment:
This is a very nice feature. Thanks for adding this!

It will also be very useful for other targets. It's not expected to test this for all different targets (there are simply too many options and we have other priorities). However, I would appreciate if you changed the code such that there is simply an assertion to check that when sample_start and sample_end are set, also otbn is selected and provide a comment where this check is done. This way we have to change much less code later on when enabling it e.g. for AES or KMAC.

Reply:
wilco